pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/ReadBridge.java

```diff
@@ -14,58 +14,81 @@
 
 import java.io.*;
 import java.nio.charset.CharacterCodingException;
+import java.util.LinkedList;
 import java.util.zip.ZipException;
 
-/*
- * ReadBridge class creates appropriate accessor and resolver.
- * It will then create the correct output conversion
- * class (e.g. Text or GPDBWritable) and get records from accessor,
- * let resolver deserialize them and reserialize them using the
- * output conversion class.
- *
- * The class handles BadRecordException and other exception type
- * and marks the record as invalid for GPDB.
+/**
+ * ReadBridge class creates appropriate accessor and resolver. It will then
+ * create the correct output conversion class (e.g. Text or GPDBWritable) and
+ * get records from accessor, let resolver deserialize them and reserialize them
+ * using the output conversion class. <br>
+ * The class handles BadRecordException and other exception type and marks the
+ * record as invalid for HAWQ.
  */
 public class ReadBridge implements Bridge {
     ReadAccessor fileAccessor = null;
     ReadResolver fieldsResolver = null;
     BridgeOutputBuilder outputBuilder = null;
+    LinkedList<Writable> outputQueue = null;
 
-    private Log Log;
+    static private Log Log = LogFactory.getLog(ReadBridge.class);
```

Comment:
best practice is to define log objects as private static final when you initialize it